crates/rpc/rpc/src/eth/api/block.rs

```diff
@@ -23,6 +23,19 @@ where
         Ok(self.client().ommers(block_id)?)
     }
 
+    pub(crate) async fn ommer_by_block_and_index(
+        &self,
+        block_id: impl Into<BlockId>,
+        index: Index,
+    ) -> EthResult<Option<RichBlock>> {
+        let block_id = block_id.into();
+        let index = usize::from(index);
+        let uncles = self.client().ommers(block_id)?.unwrap_or_default();
+        let Some(uncle_header) = uncles.get(index) else {
+            return Ok(None)
+        };
+        self.block(BlockId::from(uncle_header.number), false).await
```

Comment:
this is not correct, we need to convert he Header into a block response instead.

>  An uncle doesn't contain individual transactions.

https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getunclebyblockhashandindex

So need to construct an RPC block from header only.

Reply:
Does that mean that I should add an RPC type like `Block`, that lacks a `transactions` field and use it as a return type then? 😄

Reply:
Correct it should  be the same as get block with transaction hashes only

Reply:
I've added [this struct here](https://github.com/paradigmxyz/reth/pull/1595/files#diff-42044b02ed3f935d36786aa09df65f4d748080b9ad5b93cff782badaf9a011ebR325) for the response and a few more changes. Please, let me know if there's something else to improve :slightly_smiling_face:.